src/auditlog/context.py

```diff
@@ -0,0 +1,55 @@
+import contextlib
+import time
+import threading
+
+from django.contrib.auth import get_user_model
+from django.db.models.signals import pre_save
+from django.utils.functional import curry
+
+from auditlog.models import LogEntry
+
+threadlocal = threading.local()
+
+
+@contextlib.contextmanager
+def set_actor(actor, remote_addr=None):
+    """Connect a signal receiver with current user attached."""
+    # Initialize thread local storage
+    threadlocal.auditlog = {
+        'signal_duid': ('set_actor', time.time()),
+        'remote_addr': remote_addr,
+    }
+
+    # Connect signal for automatic logging
+    set_actor = curry(_set_actor, user=actor, signal_duid=threadlocal.auditlog['signal_duid'])
+    pre_save.connect(set_actor, sender=LogEntry, dispatch_uid=threadlocal.auditlog['signal_duid'], weak=False)
+
+    try:
+        yield
+
+    finally:
+        try:
+            auditlog = threadlocal.auditlog
+        except AttributeError:
+            pass
+        else:
+            pre_save.disconnect(sender=LogEntry, dispatch_uid=auditlog['signal_duid'])
```

Comment:
I realize that the original code did not have this, but maybe it makes sense to explicitly unset the threadlocal variable for the next use?

Reply:
Yes, it makes sense as it will make the code a bit more future-proof if any client code or new logic attempts to use `threadlocal.auditlog` for their own purposes.